### PR TITLE
Feature/oscilloscope upgrade

### PR DESCRIPTION
**Pull request by nulinspiratie**
Adds 2D plotting

### CODE REVIEW

File: qcodes/utils/oscilloscope.py [hunk 1]
```diff
@@ -209,42 +344,103 @@ def _init_qt(cls):
...
+            # if channel_settings.get("offset") is not None:
+            #     row = row + channel_settings["offset"]
+            #
+            # curve.setData(t_list_scaled, row)
+            # curve.setZValue(channel_settings.get("zorder", k))
+
+            # self.ax_2D.showGrid(x=True, y=True)
+            # self.ax_2D.disableAutoRange()
+            # self.ax_2D.setRange(xRange=(0, max(t_list_scaled)), yRange=self.ylim)
+            # self.format_axis(time_prefix=time_prefix)
```
Comment (RostyslavSavytskyy) [inline qcodes/utils/oscilloscope.py:434]: why is this here and commented out?

**Review by RostyslavSavytskyy — APPROVED** [review]
should be fine, except it's missing a bit of docstrings

**Review by RostyslavSavytskyy — APPROVED** [review]
should be good, except it's missing a bit of docstrings

File: qcodes/utils/oscilloscope.py (hunk 1 above)
Comment (nulinspiratie) [reply qcodes/utils/oscilloscope.py:434]: because it didnt work xD Removed
